app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/authentication/AuthenticationDetailsFragment.java

```diff
@@ -84,18 +124,76 @@ protected AuthenticationDetailsPresenter initPresenter() {
     @Override
     protected AuthenticationDetailsView initView() {
         return new AuthenticationDetailsViewImpl(this) {
+            @Override
+            public void logoutSuccess(AuthState authState) {
+                showMessage(R.string.logout_success);
+                if (authenticationDetailsListener != null) {
+                    authenticationDetailsListener.onLogoutSuccess(authState);
+                }
+            }
+
+            @Override
+            public void logoutFailure(Exception error) {
+                if (authenticationDetailsListener != null) {
+                    authenticationDetailsListener.onLogoutError(error);
+                }
+            }
         };
     }
 
     @Override
     public int getHelpMessageResourceId() {
-        return R.string.popup_authentication_fragment;
+        return R.string.popup_authentication_details_fragment;
     }
 
-    private void showAuthToken(Bundle args) {
-        String token = args.getString(Constants.TOKEN_FIELDS.AUTH_TOKEN);
-        if (token != null ) {
-            authResField.setText(token);
+    /**
+     * Render the Users Identity Information in the View
+     * @param args
+     */
+    private void renderIdentityInfo(Bundle args) {
+        String identityData = args.getString(Constants.TOKEN_FIELDS.IDENTITY_DATA);
+        if (identityData != null) {
+            try {
+                JSONObject identityDataJSON = new JSONObject(identityData);
```

Comment:
it's better to create a new class for the identify data, and use Gson to serialise/deserialise here.